data/services/zeppelin.py

```diff
@@ -34,19 +35,25 @@ class ZeppelinServiceOnBI(AbstractServiceOnBI):
     }
 
     def start(self):
-        print "starting zeppelin"
-        stdin, stdout, stderr = self.ssh.exec_command(self.dirName + "/bin/zeppelin-daemon.sh status")
-        zeppelin_status = stdout.read()
-        if "OK" in zeppelin_status:
-            print "Zeppelin already running. Restarting instead"
-            stdin, stdout, stderr = self.ssh.exec_command("zeppelin-0.7.0-SNAPSHOT/bin/zeppelin-daemon.sh restart")
-            print stdout.read()
-        if "FAILED" in zeppelin_status:
-            stdin, stdout, stderr = self.ssh.exec_command(self.dirName + "/bin/zeppelin-daemon.sh start")
+        # Add quick check here to see if s3AuthSet was run
+        if(self.s3authSet) : 
+
+            print "starting zeppelin"
+            stdin, stdout, stderr = self.ssh.exec_command(self.dirName + "/bin/zeppelin-daemon.sh status")
             zeppelin_status = stdout.read()
-            print "Status:  ", zeppelin_status
-            if not "OK" in zeppelin_status:
-                print stderr.read()
+            if "OK" in zeppelin_status:
+                print "Zeppelin already running. Restarting instead"
+                stdin, stdout, stderr = self.ssh.exec_command("zeppelin-0.7.0-SNAPSHOT/bin/zeppelin-daemon.sh restart")
+                print stdout.read()
+            if "FAILED" in zeppelin_status:
+                stdin, stdout, stderr = self.ssh.exec_command(self.dirName + "/bin/zeppelin-daemon.sh start")
+                zeppelin_status = stdout.read()
+                print "Status:  ", zeppelin_status
+                if not "OK" in zeppelin_status:
+                    print stderr.read()
+        else :
+            print " ERROR : need to run setS3auth method prior to running start method"
+            exit()
```

Comment:
performed some simple manual tests ...
1.  tested is s3setAuth not run prior to zeppelin start
2.  tested with rootkey.csv file missing 
both tests passed as expected